src/parkapi_sources/converters/apcoa/validators.py

```diff
@@ -0,0 +1,224 @@
+    MaxSuffix: Optional[str] = Noneable(StringValidator())
+    Currency: Optional[str] = Noneable(StringValidator())
+    CurrencyCode: Optional[str] = Noneable(StringValidator())
+    TaxRate: Optional[Decimal] = Noneable(NumericValidator())
+
+
+@validataclass
+class ApcoaParkingSiteInput:
+    CarParkId: int = IntegerValidator(allow_strings=True)
+    CarparkLongName: Optional[str] = Noneable(StringValidator())
+    CarparkShortName: Optional[str] = Noneable(StringValidator())
+    CarParkWebsiteURL: Optional[str] = Noneable(UrlValidator())
+    CarParkPhotoURLs: Optional[ApcoaCarparkPhotoURLInput] = Noneable(DataclassValidator(ApcoaCarparkPhotoURLInput))
+    CarparkType: ApcoaCarparkTypeNameInput = DataclassValidator(ApcoaCarparkTypeNameInput)
+    Address: ApcoaAdressInput = DataclassValidator(ApcoaAdressInput)
+    NavigationLocations: list[ApcoaNavigationLocationsInput] = ListValidator(DataclassValidator(ApcoaNavigationLocationsInput))
+    Spaces: list[ApcoaParkingSpaceInput] = ListValidator(DataclassValidator(ApcoaParkingSpaceInput))
+    OpeningHours: list[ApcoaOpeningHoursInput] = ListValidator(DataclassValidator(ApcoaOpeningHoursInput))
+    LastModifiedDateTime: datetime = SpacedDateTimeValidator(
+        local_timezone=ZoneInfo('Europe/Berlin'),
+        target_timezone=timezone.utc,
+    )
+    IndicativeTariff: Optional[ApcoaIndicativeTariffInput] = Noneable(DataclassValidator(ApcoaIndicativeTariffInput))
+
+    # TODO: ignored multiple attributes which do not matter so far
+
+    def __post_init__(self):
+        for capacity in self.Spaces:
+            if capacity.Type == ApcoaParkingSpaceType.TOTAL_SPACES:
+                return
+        # If no capacity with type PARKING was found, we miss the capacity and therefore throw a validation error
+        raise ValidationError(reason='Missing parking spaces capacity')
+
+    def get_osm_opening_hours(self) -> str:
+        apcoa_opening_times: dict[str, list[str]] = defaultdict(list)
+        for opening_hours_input in self.OpeningHours:
+            opening_time = opening_hours_input.OpeningTimes.replace(' ', '')
+            if opening_hours_input.OpeningTimes == '00:00 - 00:00':
+                opening_time = '00:00-24:00'
+                apcoa_opening_times['24/7'].append(opening_time)
+            if opening_hours_input.Weekday in list(ApcoaOpeningHoursWeekday)[:5] and opening_time != 'closed':
+                apcoa_opening_times['Mo-Fr'].append(opening_time)
+            if opening_hours_input.Weekday in list(ApcoaOpeningHoursWeekday) and opening_time != 'closed':
+                apcoa_opening_times[opening_hours_input.Weekday.value].append(opening_time)
+
+        osm_opening_hour: list = []
+        if len(apcoa_opening_times['Mo-Fr']) == 5 and len(set(apcoa_opening_times['Mo-Fr'])) == 1:
+            osm_opening_hour.append(f'Mo-Fr {next(iter(set(apcoa_opening_times["Mo-Fr"])))}')
+        else:
+            if 'Monday' in list(apcoa_opening_times.keys()):
```

Comment:
Why not iterating over list(ApcoaOpeningHoursWeekday)?

``` 
for weekday in list(ApcoaOpeningHoursWeekday)[:5]:
  print(weekday.name) -> MONDAY
  print(weekday.value) -> Monday
  print(weekday.to_osm_opening_day_format()) -> Mo
```

Reply:
good tip, thanks!